### PR TITLE
Added Image Processing Algorithms

### PR DESCRIPTION
**Pull request by MuLx10**
Added Image Processing algorithms Currently it contains modules filters and transformations.

### CODE REVIEW

File: code/artificial_intelligence/src/image_processing/Canny/canny.cpp [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+#include <iostream>
+#define _USE_MATH_DEFINES
+#include <cmath>
+#include <vector>
+#include "canny.h"
+#include "opencv2/imgproc/imgproc.hpp"
+#include "opencv2/highgui/highgui.hpp"
+
+using namespace std;
```
Comment (arnavb) [inline code/artificial_intelligence/src/image_processing/Canny/canny.cpp]: Could you not use either of these?

**Comment by codezoned** [comment]
Filters also come into Image processing so It would be better if you break them down into simpler terms to add @MuLx10

**Comment by MuLx10** [comment]
@arnab I have to use it, @codezoned Can you please explain. I didn't get

**Comment by MuLx10** [comment]
@arnavb  for image matrix manipulation

**Comment by MuLx10** [comment]
okay @arnavb,  I mistook it to be library includes.I'll do it .

**Comment by arnavb** [comment]
@MuLx10 Both lines are still there.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**
positive_opengenus_first

**Pull request merged**